src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/Message.java

```diff
@@ -25,19 +20,14 @@
     @Setter
     private Interchange interchange;
 
-    @Getter
-    @Setter
-    private List<Transaction> transactions;
-
     public Message(final List<String> edifactSegments) {
         super(edifactSegments);
     }
 
     public String findFirstGpCode() {
-        return getTransactions().stream()
+        return extractOptionalSegment(GpNameAndAddress.KEY_QUALIFIER)
+            .stream()
             .limit(1)
```

Comment:
A stream from an optional can only have at most 1 element anyway, so `.limit(1)` is now redundant.

Reply:
Thanks for letting me know this was redundant 😄  

Have removed it in 697ebb8